src/unit-taxonomy-tags-drawer/UnitTaxonomyTagsDrawer.jsx

```diff
@@ -0,0 +1,118 @@
+  };
+
+  const useUnitTaxonomyTagsData = () => {
+    const unitTaxonomyTagsData = useUnitTaxonomyTagsDataResponse(unitId);
+    const isUnitTaxonomyTagsLoaded = useIsUnitTaxonomyTagsDataLoaded(unitId);
+    return { unitTaxonomyTagsData, isUnitTaxonomyTagsLoaded };
+  };
+
+  const useTaxonomyListData = () => {
+    const taxonomyListData = useTaxonomyListDataResponse(org);
+    const isTaxonomyListLoaded = useIsTaxonomyListDataLoaded(org);
+    return { taxonomyListData, isTaxonomyListLoaded };
+  };
+
+  const { unitData, isUnitDataLoaded } = useUnitData();
+  const { unitTaxonomyTagsData, isUnitTaxonomyTagsLoaded } = useUnitTaxonomyTagsData();
+  const { taxonomyListData, isTaxonomyListLoaded } = useTaxonomyListData();
+
+  const closeUnitTaxonomyTagsDrawer = () => {
+    // "*" allows communication with any origin
+    window.parent.postMessage('closeManageTagsDrawer', '*');
+  };
+
+  useEffect(() => {
+    const handleEsc = (event) => {
+      /* Close drawer when ESC-key is pressed and selectable dropdown box not open */
+      const targetElement = event.target;
+      const selectableBox = targetElement.classList.contains('pgn__selectable_box');
+      if (event.key === 'Escape' && !selectableBox) {
+        closeUnitTaxonomyTagsDrawer();
+      }
+    };
+    document.addEventListener('keydown', handleEsc);
+
+    return () => {
+      document.removeEventListener('keydown', handleEsc);
+    };
+  }, []);
+
+  const taxonomies = useMemo(() => {
+    if (taxonomyListData && unitTaxonomyTagsData) {
+      // Initialize list of unit tags in taxonomies to populate
+      const taxonomiesList = taxonomyListData.results.map((taxonomy) => {
+        // eslint-disable-next-line no-param-reassign
+        taxonomy.unitTags = [];
+        return taxonomy;
+      });
+
+      const unitTaxonomies = unitTaxonomyTagsData.taxonomies;
+
+      // eslint-disable-next-line array-callback-return
+      unitTaxonomies.map((unitTaxonomyTags) => {
+        const unitTaxonomy = taxonomiesList.find((taxonomy) => taxonomy.id === unitTaxonomyTags.taxonomy_id);
+        if (unitTaxonomy) {
+          unitTaxonomy.unitTags = unitTaxonomyTags.tags;
+        }
+      });
+
+      return taxonomiesList;
+    }
+    return [];
+  }, [taxonomyListData, unitTaxonomyTagsData]);
+
+  return (
+
+    <div className="mt-1">
+      <Container size="xl">
+        <CloseButton onClick={() => closeUnitTaxonomyTagsDrawer()} data-testid="drawer-close-button" />
+        <span>{intl.formatMessage(messages.headerSubtitle)}</span>
+        <h3>{ isUnitDataLoaded ? unitData.displayName : 'loading...' }</h3>
+        <hr />
+
+        { isTaxonomyListLoaded && isUnitTaxonomyTagsLoaded
+          ? taxonomies.map((data) => (
+            <div key={`taxonomy-tags-collapsible-${data.id}`}>
+              <TaxonomyTagsCollapsible taxonomyAndTagsData={data} />
+              <hr />
+            </div>
+          ))
+          : <h3>Loading taxonomy data...</h3> }
```

Comment:
Just use the `<Loading />` component to render a spinner instead of displaying this text (search the code for other examples). Otherwise, we'd have to translate this string.

Reply:
I used a mix of the `<Loading />` and the `<Spinner />` components since I had issues with getting the styling to show both of them. `<Loading />` takes over the whole height of the screen.

Reply:
Oh ok, sure. Whatever works! I'm actually not too familiar with these components; I just knew there should be some existing component that doesn't need text. A spinner itself is totally fine.